### ISSUE TITLE
Attention Logit Metrics Captured Post-Softmax Instead of Pre-Softmax

### ISSUE THREAD

**Issue by Aditya8369** [l3tchupkt/adaptq#135]
File: runtime/runtime_context.cpp:L370-L376
Description: In RuntimeContext::compute(), logits_ is modified in-place during softmax probability calculation:
```
/* Softmax */
for (int i = 0; i < n; ++i) { logits_[i] = expf(logits_[i] - mx); sv += logits_[i]; }
float inv_s = 1.f / sv;
for (int i = 0; i < n; ++i) logits_[i] *= inv_s;
m.logit_max = *std::max_element(logits_.begin(), logits_.begin() + n);
m.logit_min = *std::min_element(logits_.begin(), logits_.begin() + n);
```
m.logit_max and m.logit_min end up recording the maximum and minimum post-softmax probabilities (bounded between $0.0$ and $1.0$) instead of the actual pre-softmax attention logit scores.
Steps to Fix:
Open runtime/runtime_context.cpp
.
Extract m.logit_max and m.logit_min before the softmax loop:
cpp
m.logit_max = mx;
m.logit_min = *std::min_element(logits_.begin(), logits_.begin() + n);
